mutiny-core/src/lib.rs

```diff
@@ -2189,6 +2197,46 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    /// Calls upon a Cashu mint and withdrawls/redeem/melts the token from it.
+    pub async fn melt_cashu_token(&self, token_v3: TokenV3) -> Result<bool, MutinyError> {
```

Comment:
I think it'd make more sense if this function returned `MutinyInvoice`, can just call `self.get_invoice(mutiny_invoice_str)` at the end to get it.

if it doesn't end up paid can just return an error like `MutinyError::RoutingFailed` or something

Reply:
ok, I think a `Vec<MutinyInvoice>`? Since we are looping through tokens and could generate multiple invoices.

Reply:
Ah, is there no way to aggregate them? Could we not reissue them into a single token then do one withdrawal

Reply:
Oh interesting. Okay then yeah `Vec<MutinyInvoice>` makes sense to me